R/include-podiatrist-services.R

```diff
@@ -0,0 +1,57 @@
+#' Include diabetes-specific podiatrist services.
+#'
+#' #' Uses the `sysi` or `sssy` registers as input to extract the dates of all
+#' diabetes-specific podiatrist services. Removes duplicate services on the
+#' same date. Only the two earliest dates per individual are kept.
+#'
+#' #' The output is passed to the `join_inclusions()` function for the final
+#' step of the inclusion process.
+#'
+#' @param sysi The SYSI register.
+#' @param sssy The SSSY register.
+#'
+#' @return The same type as the input data, default as a [tibble::tibble()],
+#'   with two columns and up to two rows for each individual:
+#'
+#'   -  `pnr`: Identifier variable
+#'   -  `date`: The dates of the first and second diabetes-specific
+#'      podiatrist record
+#'  -  `has_podiatrist_services`: A logical variable that acts as a helper
+#'      indicator for use in later functions.
+#'
+#' @keywords internal
+#' @inherit algorithm seealso
+#'
+#' @examples
+#' \dontrun{
+#' include_podiatrist_services(register_data$sysi, register_data$sssy)
+#' }
+include_podiatrist_services <- function(sysi, sssy) {
+  verify_required_variables(sysi, "sysi")
+  verify_required_variables(sssy, "sssy")
+  criteria <- get_algorithm_logic("podiatrist_services") |>
+    # To convert the string into an R expression.
+    rlang::parse_expr()
+
+  column_names_to_lower(sysi) |>
+    dplyr::full_join(column_names_to_lower(sssy),
+      by = dplyr::join_by(pnr, barnmak, speciale, honuge)
+    ) |>
```

Comment:
After explicitly adding the columns to join by, I get a note from `devtools::check()`:
<img width="517" alt="image" src="https://github.com/user-attachments/assets/1751e091-a17a-48f4-96b1-f4779caaa3ec" />

I'm not sure how to fix this, since `join_by()` doesn't seem to allow the use of `.data$`